File: Month 1/Android/intro_to_jetpack/app/src/main/java/com/example/introtojetpack/activities/MainActivity.java
package com.example.introtojetpack.activities;


import android.arch.lifecycle.ViewModelProviders;
import android.graphics.Color;
import android.support.constraint.ConstraintLayout;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.example.introtojetpack.R;
import com.example.introtojetpack.fragments.RecyclerFragment;
import com.example.introtojetpack.viewmodel.MyViewModel;

import java.util.Random;


public class MainActivity extends AppCompatActivity {
    private MyViewModel mModel;
    private ConstraintLayout mChangeColor;
    private FragmentManager mFragmentManager = getSupportFragmentManager();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mChangeColor =findViewById(R.id.colorchange);
        mModel = ViewModelProviders.of(this).get(MyViewModel.class);

        findViewById(R.id.bt_color_change).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getColor();
            }
        });


        mChangeColor.setBackgroundColor(mModel.saveColor);

        RecyclerFragment recyclerFragment = new RecyclerFragment();
        FragmentTransaction fragmentTransaction = mFragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.recyclerView_replacing,recyclerFragment);
        fragmentTransaction.commit();


    }

    void getColor(){

        Random random=new Random();
        int color= Color.argb(255,random.nextInt(256),random.nextInt(256),random.nextInt(256));
        mChangeColor.setBackgroundColor(color);
        mModel.saveColor = color;

    }


}

File: Bootcamp/Android/intro to kotlin 4/KotlinLogin/app/src/main/java/com/example/kotlinlogin/MainActivity.kt
package com.example.kotlinlogin

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.TextUtils
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var etName = findViewById<EditText>(R.id.et_name)
        var etPass = findViewById<EditText>(R.id.et_pass)

        findViewById<Button>(R.id.bt_login).setOnClickListener {

            if (!TextUtils.isEmpty(etName.getText().toString()) && !TextUtils.isEmpty(etPass.getText().toString())) {
                intent = Intent(this@MainActivity, AfterLogin::class.java)
                startActivity(intent)
            } else {
                Toast.makeText(this, "Fill all Details", Toast.LENGTH_SHORT).show()
            }

        }

    }

}
